File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    public int climbStairs(int n) {
        if (n == 0) return 0;
        if (n == 1) return 1;
        if (n == 2) return 2;

        int x = 1;
        int y = 2;
        int result = 3;
        
        for (int i = 3; i <= n; i++) {
            result = x + y;
            x = y;
            y = result;
        }

        return result;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        result = [-1] * k

        frequencies = {}
        for i in nums:
            frequencies[i] = 1 + frequencies.get(i, 0)

        i = 0
        n = 0
        greatest = 0
        for k in frequencies:
            if frequencies[k] > greatest:
                result[i] = k
                i = (i + 1) % k
        
        print(result)

        return result
